discovery/discovery.go

```diff
@@ -55,12 +59,65 @@ func (o *orchestratorPool) GetInfos() []common.OrchestratorLocalInfo {
 	return o.infos
 }
 
+func ResolveURL(urlToResolve *url.URL) []*url.URL {
+	resolvedUrls := make([]*url.URL, 0, 0)
+	// check if there's a port in host, for ipv6 it's after ]
+	v6PortSplit := strings.Split(urlToResolve.Host, "]")
+	hasPort := strings.Count(v6PortSplit[len(v6PortSplit)-1], ":") == 1
+	host, port := "", ""
+	if hasPort {
+		host, port, _ = net2.SplitHostPort(urlToResolve.Host)
+	} else {
+		host, port = urlToResolve.Host, ""
+	}
+	if net2.ParseIP(host) == nil {
+		// handle localhost - won't work for custom hosts entry!
+		var addrs []string
+		var err error
+		if strings.ToLower(host) == "localhost" {
+			// ipv6 addr may be abbreviated or not
+			addrs = []string{"127.0.0.1", "::1", "0000:0000:0000:0000:0000:0000:0000:0001"}
+		} else {
+			addrs, err = hostLookupFn(host)
+		}
+		if err != nil {
+			clog.Errorf(context.TODO(), "Couldn't resolve domain for URL %s, check local DNS settings. Discovery may not work properly.", urlToResolve.String())
+			return append(resolvedUrls, urlToResolve)
+		}
+		for _, addr := range addrs {
+			newUrl, _ := url.Parse(urlToResolve.String())
+			if port != "" {
+				// enclose ipv6 into brackets
+				if strings.Count(addr, ":") > 0 {
+					newUrl.Host = fmt.Sprintf("[%s]:%s", addr, port)
+				} else {
+					newUrl.Host = fmt.Sprintf("%s:%s", addr, port)
+				}
+			} else {
+				newUrl.Host = addr
+			}
+			resolvedUrls = append(resolvedUrls, newUrl)
+		}
+	} else {
+		// already ip or error, do nothing
+		return append(resolvedUrls, urlToResolve)
+	}
+	return resolvedUrls
+}
+
 func (o *orchestratorPool) GetInfo(uri string) common.OrchestratorLocalInfo {
 	var res common.OrchestratorLocalInfo
+	queryUrl, _ := url.Parse(uri)
+	resolvedUrls := ResolveURL(queryUrl)
 	for _, info := range o.infos {
-		if info.URL.String() == uri {
-			res = info
-			break
+		oResolvedUrls := ResolveURL(info.URL)
+		for _, oUrl := range oResolvedUrls {
+			for _, queryUrl := range resolvedUrls {
+				if oUrl.String() == queryUrl.String() {
```

Comment:
```suggestion
			for _, qUrl := range resolvedUrls {
				if oUrl.String() == qUrl.String() {
```

Nit: you already have the `queryUrl` defined.

Reply:
Fixed.